dask/dataframe/core.py

```diff
@@ -337,8 +593,47 @@ def __init__(self, dsk, name, meta, divisions):
                 f"Expected meta to specify type {type(self).__name__}, got type "
                 f"{typename(type(meta))}"
             )
-        self._meta = meta
-        self.divisions = tuple(divisions)
+
+        # Set partition metadata
+        if partition_metadata is None:
+            partition_metadata = PartitionMetadata(
+                meta=meta,
+                divisions=tuple(divisions),
+            )
+        else:
+            partition_metadata.set_meta(meta)
+            partition_metadata.set_divisions(divisions)
+        self.set_partition_metadata(partition_metadata)
+
+    @property
+    def partition_metadata(self):
+        return self._partition_metadata
+
+    def set_partition_metadata(self, value):
+        if isinstance(value, PartitionMetadata):
+            self._partition_metadata = value
+        else:
+            raise ValueError(f"Expected PartitionMetadata object, got {type(value)}")
```

Comment:
Here and elsewhere, is there a reason you've opted to use `set_*` rather than a property setter?